### PR TITLE
CR TTT

### PR DESCRIPTION
**Pull request by XanderJS**
Code review tic tac toe.

### CODE REVIEW

File: tictactoe.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    set var="title"     value="Tic Tac Toe"             --><!--#
+    set var="header"    value="Tic Tac Toe"             --><!--#
+    set var="subheader" value="Have fun!"               --><!--#
+    set var="tictactoe" value="active"                  --><!--#
+    set var="link"      value="tictactoe/tictactoe.css" --><!--#
+
+    include virtual="ssi/header.html" --><!--#
+    include virtual="ssi/nav.html"    -->
+
+<article>
```
Comment (gryphonshafer) [inline tictactoe.html:11]: I recommend using an `id` here to label the entire TTT component. Then switch `id` to `class` below within the HTML of this component.

File: tictactoe.html [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    <p id="player">
+        X
+    </p>
+
+    <div id="main">
+        <table id="game">
+            <tbody>
+                <tr>
+                    <td class="cell" id="0"></td>
```
Comment (gryphonshafer) [inline tictactoe.html:25]: All these `class="cell"` that exist within the `<tbody>` block are redundant. You're repeating yourself. DRY. Remove all the `class="cell"` and instead use a CSS selector like: `article#ttt table td`

File: tictactoe.html [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                </tr>
+            </tbody>
+        </table>
+    </div>
+
+    <p id="endgame">
+    </p>
+
+    <p id="restart">
+        <button id="reset" onClick="startGame()">Restart</button>
```
Comment (gryphonshafer) [inline tictactoe.html:47]: The `onClick` shouldn't exist in the HTML. It should exist in the JS.

File: tictactoe/tictactoe.css [hunk 4]
```diff
@@ -0,0 +1,76 @@
+article {
+    margin      : 0;
+    line-height : 0.5;
+}
+
+#next {
```
Comment (gryphonshafer) [inline tictactoe/tictactoe.css:6]: Of course, you'll need to change your accessors based on your HTML `id` changes.

File: tictactoe/tictactoe.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
+var origBoard;
```
Comment (gryphonshafer) [inline tictactoe/tictactoe.js:1]: `var` should probably be `let`, but before you make the change, read about the differences and make a wise decision...

File: tictactoe/tictactoe.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [2,4,6],
+]
+const cells = document.querySelectorAll(".cell");
+startGame();
+
+function startGame() {
+    document.querySelector("#endgame").style.display = "none";
```
Comment (gryphonshafer) [inline tictactoe/tictactoe.js:19]: I'd place startup state style in the CSS, then put runtime style changes into the JS.

File: tictactoe/tictactoe.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    for (var i = 0; i < cells.length; i++) {
+        cells[i].innerText = "";
+        cells[i].style.removeProperty("background-color");
+        cells[i].addEventListener("click", turnClick, false);
+    }
+}
+
+function turnClick(square) {
+    if (square.target.innerText == "") {
+        turn(square.target.id, player)
```
Comment (gryphonshafer) [inline tictactoe/tictactoe.js:33]: Missing `;`

File: tictactoe/tictactoe.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    [2,4,6],
+]
+const cells = document.querySelectorAll(".cell");
+startGame();
+
+function startGame() {
+    document.querySelector("#endgame").style.display = "none";
+    document.querySelector("#restart").style.display = "none";
+    player = player1 ;
+    document.getElementById("player").innerHTML = "X" ;
```
Comment (gryphonshafer) [inline tictactoe/tictactoe.js:22]: The value should be `player` instead of `"X"`, right?

File: tictactoe/tictactoe.js [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    origBoard = Array.from(Array(9).keys());
+    for (var i = 0; i < cells.length; i++) {
+        cells[i].innerText = "";
+        cells[i].style.removeProperty("background-color");
```
Comment (gryphonshafer) [inline tictactoe/tictactoe.js:26]: What does this line do?

File: tictactoe/tictactoe.js [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+        cells[i].addEventListener("click", turnClick, false);
+    }
+}
+
+function turnClick(square) {
+    if (square.target.innerText == "") {
+        turn(square.target.id, player)
+        player == player1 ? player = player2 : player = player1;
+        player == player1 ?
```
Comment (gryphonshafer) [inline tictactoe/tictactoe.js:35]: Improperly formatted ternary

File: tictactoe/tictactoe.js [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+        document.getElementById("player").innerHTML = "X" :
```
Comment (gryphonshafer) [inline tictactoe/tictactoe.js:36]: Values here should be based on `player1` and `player2` so you only define "X" and "Y" in a single place. DRY. This allows you to change to "A" and "B" if you want with trivial effort.

File: tictactoe/tictactoe.js [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+        document.getElementById("player").innerHTML = "O" ;
+    }
+}
+
+function turn(squareId, player) {
+    origBoard[squareId] = player;
+    document.getElementById(squareId).innerText = player;
+    let gameWon = checkWin(origBoard, player)
```
Comment (gryphonshafer) [inline tictactoe/tictactoe.js:44]: Missing `;`

File: tictactoe/tictactoe.js [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+    if (gameWon) {
+        gameOver(gameWon)
```
Comment (gryphonshafer) [inline tictactoe/tictactoe.js:46]: `;`?

File: tictactoe/tictactoe.js [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+        }
+        declareWinner("Tie")
+    }
+}
+
+function gameOver(gameWon) {
+    for (let index of winCombos[gameWon.index]) {
+        document.getElementById(index).style.backgroundColor =
+        gameWon.player == player1 ? "#09f" : "#f00";
+        gameWon.player == player1 ? declareWinner("X Wins!") : declareWinner("O Wins!") ;
```
Comment (gryphonshafer) [inline tictactoe/tictactoe.js:83]: `declareWinner( gameWon.player + " Wins!");`

File: tictactoe/tictactoe.js [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+        checkTie();
+    }
+}
+
+function emptySquares() {
+    return origBoard.filter(s => typeof s == "number");
+}
+
+function checkWin(board, player) {
+    let plays = board.reduce((a, e, i) =>
```
Comment (gryphonshafer) [inline tictactoe/tictactoe.js:58]: This line isn't long enough to justify a line break, is it?

File: tictactoe/tictactoe.js (hunk 8 above)
Comment (XanderJS) [reply tictactoe/tictactoe.js:22]: Not here. This tells the program to show that it's X's turn at the start and when the game is reset.

File: tictactoe/tictactoe.js (hunk 9 above)
Comment (XanderJS) [reply tictactoe/tictactoe.js:26]: These lines take all of the tic tac toe squares and puts them back to having nothing in them.

File: tictactoe/tictactoe.js (hunk 10 above)
Comment (XanderJS) [reply tictactoe/tictactoe.js:35]: What do you mean by that?

File: tictactoe/tictactoe.js (hunk 6 above)
Comment (XanderJS) [reply tictactoe/tictactoe.js:19]: This is so when you click the reset button, it hides the winner data and the reset button from view.

File: tictactoe/tictactoe.js (hunk 11 above)
Comment (XanderJS) [reply tictactoe/tictactoe.js:36]: How should I go about doing that?

Comment (gryphonshafer) [reply tictactoe/tictactoe.js:36]: Just use `player1` instead of "X"

**Pull request closed without merging**